src/argilla/server/apis/v1/handlers/responses.py

```diff
@@ -49,36 +51,19 @@ async def _get_response(db: AsyncSession, response_id: UUID) -> Response:
     return response
 
 
-async def _get_record(db: AsyncSession, record_id: UUID) -> Record:
-    record = await datasets.get_record_by_id(db, record_id, with_dataset=True)
-    if record is None:
-        raise errors.NotFoundError(f"Record with id `{record_id}` not found")
-
-    return record
-
-
 @router.post("/me/responses/bulk", response_model=ResponsesBulk)
 async def create_current_user_responses_bulk(
     *,
-    db: AsyncSession = Depends(get_async_db),
-    search_engine: SearchEngine = Depends(get_search_engine),
     body: ResponsesBulkCreate,
     current_user: User = Security(auth.get_current_user),
+    use_case: UpsertResponsesInBulkUseCase = Depends(UpsertResponsesInBulkUseCaseFactory()),
 ):
-    responses_bulk_items = []
-    for item in body.items:
-        try:
-            record = await _get_record(db, item.record_id)
-
-            await authorize(current_user, RecordPolicyV1.create_response(record))
-
-            response = await datasets.upsert_response(db, search_engine, record, current_user, item)
-        except Exception as err:
-            responses_bulk_items.append(ResponseBulk(item=None, error=ResponseBulkError(detail=str(err))))
-        else:
-            responses_bulk_items.append(ResponseBulk(item=ResponseSchema.from_orm(response), error=None))
-
-    return ResponsesBulk(items=responses_bulk_items)
+    try:
+        responses_bulk_items = await use_case.execute(body.items, user=current_user)
+    except NotFoundError as err:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=str(err))
+    else:
+        return ResponsesBulk(items=responses_bulk_items)
```

Comment:
```suggestion
return ResponsesBulk(items=responses_bulk_items)
```
I'm an else hater, feel free to apply or discard

Reply:
I think the same, but this case makes clearer the `else` case when nothing unexpected occurs. I will keep it as is.